crates/league-toolkit/src/core/wad/builder.rs

```diff
@@ -0,0 +1,262 @@
+            // Write the header
+            writer.write_u16::<LE>(0x5752)?;
+            writer.write_u8(3)?; // major
+            writer.write_u8(4)?; // minor
+
+            // Write dummy ECDSA signature
+            writer.write_all(&[0; 256])?;
+            writer.write_u64::<LE>(0)?;
+
+            // Write dummy TOC
+            writer.write_u32::<LE>(self.chunk_builders.len() as u32)?;
+            let toc_offset = writer.stream_position()?;
+            for _ in self.chunk_builders.iter() {
+                writer.write_all(&[0; 32])?;
+            }
+
+            Ok::<_, WadBuilderError>(toc_offset)
+        })?;
+
+        Ok((header_toc_size, toc_offset))
+    }
+
+    fn compress_chunk_data(
+        data: &[u8],
+        force_compression: Option<WadChunkCompression>,
+    ) -> Result<(Vec<u8>, WadChunkCompression), WadBuilderError> {
+        let (compressed_data, compression) = match force_compression {
+            Some(compression) => (
+                Self::compress_chunk_data_by_compression(data, compression)?,
+                compression,
+            ),
+            None => {
+                let kind = LeagueFileKind::identify_from_bytes(data);
+                let compression = kind.ideal_compression();
+                let compressed_data = Self::compress_chunk_data_by_compression(data, compression)?;
+
+                (compressed_data, compression)
+            }
+        };
+
+        Ok((compressed_data, compression))
+    }
+
+    fn compress_chunk_data_by_compression(
+        data: &[u8],
+        compression: WadChunkCompression,
+    ) -> Result<Vec<u8>, WadBuilderError> {
+        let mut compressed_data = Vec::new();
+        match compression {
+            WadChunkCompression::None => {
+                compressed_data = data.to_vec();
+            }
+            WadChunkCompression::GZip => {
+                let reader = BufReader::new(data);
+                let mut encoder = GzEncoder::new(reader, flate2::Compression::default());
+
+                encoder.read_to_end(&mut compressed_data)?;
+            }
+            WadChunkCompression::Zstd => {
+                let mut encoder = zstd::Encoder::new(BufWriter::new(&mut compressed_data), 3)?;
+                encoder.write_all(data)?;
+                encoder.finish()?;
+            }
+            WadChunkCompression::Satellite => {
+                return Err(WadBuilderError::UnsupportedCompressionType(compression));
+            }
+            WadChunkCompression::ZstdMulti => {
+                return Err(WadBuilderError::UnsupportedCompressionType(compression));
+            }
+        }
+
+        Ok(compressed_data)
+    }
+}
+
+/// Implements a builder interface for creating WAD chunks.
+///
+/// # Examples
+/// ```
+/// # use league_toolkit::core::wad::*;
+/// #
+/// let builder = WadChunkBuilder::default();
+/// builder.with_path("path/to/chunk");
+/// builder.with_force_compression(WadChunkCompression::Zstd);
```

Comment:
Is there any easy way to make this also act like a useful doctest? Not super important but would be nice

Reply:
Could you elaborate ?
One thing I can say is that I had to fix the docs code because `cargo test` was failing